torch/_higher_order_ops/wrap.py

```diff
@@ -236,5 +239,8 @@ def __call__(self, gmod, *args, **kwargs):
             with fx_traceback.preserve_node_meta():
                 return Interpreter(gmod).run(*args)
 
+    def pure(self):
```

Comment:
I think we should call this something different like `cacheable` or `is_cache_compatible`. `pure` is too strong of definition here, for example `cond` is not necessarily pure if the operations that are passed to it are not pure, but cond itself is cacheable if its arguments are cacheable.